cdc/sink/codec/canal_flat.go

```diff
@@ -261,14 +278,21 @@ func (c *CanalFlatEventBatchEncoder) newFlatMessageForDML(e *model.RowChangedEve
 		SQLType:       sqlType,
 		MySQLType:     mysqlType,
 		Data:          make([]map[string]interface{}, 0),
-		Old:           make([]map[string]interface{}, 0),
+		Old:           nil,
 		tikvTs:        e.CommitTs,
 	}
 
-	// We need to ensure that both Data and Old have exactly one element,
-	// even if the element could be nil. Changing this could break Alibaba's adapter
-	flatMessage.Data = append(flatMessage.Data, data)
-	flatMessage.Old = append(flatMessage.Old, oldData)
+	if e.IsDelete() {
+		flatMessage.Data = append(flatMessage.Data, oldData)
+	} else if e.IsInsert() {
+		flatMessage.Data = append(flatMessage.Data, data)
+	} else if e.IsUpdate() {
+		flatMessage.Old = make([]map[string]interface{}, 0)
+		flatMessage.Old = append(flatMessage.Old, oldData)
```

Comment:
Can you combine the lines and use a map literal? It would be more readable this way.